tests/openthread/main.c

```diff
@@ -0,0 +1,216 @@
+    otMessage *message;
+
+    otUdpSocket socket;
+
+    memset(&socket, 0, sizeof(otUdpSocket));
+
+    otUdpOpen(ot_instance, &socket, _handle_receive, NULL);
+
+    message = otUdpNewMessage(ot_instance, true);
+     otMessageSetLength(message, pkt->len);
+    otMessageWrite(message, 0, pkt->payload, pkt->len);
+
+    otMessageInfo mPeer;
+
+    /* Set dest address */
+    memcpy(&mPeer.mPeerAddr.mFields, &(pkt->ip_addr), sizeof(ipv6_addr_t));
+
+    /* Set dest port */
+    mPeer.mPeerPort = pkt->port;
+
+    otUdpSend(&socket, message, &mPeer);
+    otUdpClose(&socket);
+}
+
+int _udp(int argc, char **argv)
+{
+    if (argc < 3) {
+		printf("Usage: udp server <port>\n")
+		printf("       udp send <ip_addr> <port> <message>\n")
+        return 1;
+    }
+    else if (strcmp(argv[1], "server") == 0) {
+        uint16_t port = atoi(argv[2]);
+        ot_exec_job(_create_udp_socket, &port);
+    }
+    else if (argc >= 2 && strcmp(argv[1], "send") == 0) {
+        /* send packet */
+        udp_pkt_t pkt;
+        ipv6_addr_from_str(&pkt.ip_addr, argv[2]);
+        pkt.port = atoi(argv[3]);
+        pkt.payload = argv[4];
+        pkt.len = strlen(argv[4]);
+        ot_exec_job(_send_udp_pkt, &pkt);
+    }
+    return 0;
+}
+
+static void _usage(char *cmd)
+{
+    printf("Usage: %s \n", cmd);
+    puts("        List Thread IPv6 addresses");
+    printf("    %s <get|set> <panid> \n", cmd);
+    puts("        Get/Set PAN ID");
+    printf("    %s thread start \n", cmd);
+    puts("        Start the thread network");
+    printf("    %s thread state \n", cmd);
+    puts("        Asks for the state of a thread network.");
+}
+
+int _ifconfig(int argc, char **argv)
+{
+    uint16_t panid;
+
+    if (argc >= 2) {
+        if (strcmp(argv[1], "get") == 0 && argc >= 3) {
+            if (strcmp(argv[2], "panid") == 0) {
+                ot_exec_job(_get_panid, &panid);
+            }
+        }
+        else if (strcmp(argv[1], "set") == 0 && argc >= 4) {
+            if (strcmp(argv[2], "panid") == 0) {
+                panid = strtol(argv[3], NULL, 0);
+                ot_exec_job(_set_panid, &panid);
+            }
+        }
+        else if (strcmp(argv[1], "thread") == 0) {
+            if (strcmp(argv[2], "start") == 0) {
+                ot_exec_job(_thread_start, NULL);
+            }
+            else if (strcmp(argv[2], "state") == 0) {
+                uint8_t state;
+                ot_exec_job(_read_state, &state);
+            }
+        }
+        else {
+            _usage(argv[0]);
+        }
+    }
+    else {
+        ot_exec_job(_get_ip_addresses, NULL);
+    }
+    return 0;
+}
+
+#if !defined(MODULE_OPENTHREAD_CLI) && !defined(MODULE_OPENTHREAD_NCP)
+static const shell_command_t shell_commands[] = {
+    { "ifconfig", "Get or set panid", _ifconfig },
+    { "udp", "Test udp", _udp },
+    { NULL, NULL, NULL }
+};
+#endif
+
+int main(void)
+{
+#if defined(MODULE_OPENTHREAD_CLI) || defined(MODULE_OPENTHREAD_NCP)
```

Comment:
ok, so if I understand correctly, this PR only provides the ifconfig and udp commands provided in this main.c file ?